### ISSUE TITLE
projects/core/howto/basics.rst should be properly formatted

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6947]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#6947|
|Type|task|
|Created|2014-02-06 11:09:02Z|
|Branch|https://github.com/twisted/twisted/tree/basic-docs-format-6947-2|

As with [#6943](https://github.com/twisted/twisted/issues/6943), this is updating the rST sources to be correctly formatted.

<details><summary>Searchable metadata</summary>

```
trac-id__6947 6947
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__Lore_to_Sphinx Lore to Sphinx
branch__branches_basic_docs_format_6947_2 branches/basic-docs-format-6947-2
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1391684942000000 1391684942000000
changetime__1410787360593113 1410787360593113
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [41567]) Branching to basic-docs-format-6947.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Putting this up for review.

Not sure if the comments were required (the first looks to be something weird that's not required, the other is not needed as note is the correct thing to be using there).

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

Looks good.

Maybe ``` (usually you would do kill `cat twistd.pid`).``` can be put in a block to make it easier to spot what is the exact command

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_4"></a>|@hynek set owner to @hawkowl|
|-|-|

1.13: Please make this an enumeration.
 1. 14: Wasn’t there some talk about writing out namespaces on first mention? I honestly can’t remember but it seems to me like that. :)
 1. 16: We don’t really capitalize HOWTO like that, do we?
 1. 17: This comma irritates me, any reason to not put a colon there?
 1. 19: `````Application`` s``` should be `````Application``\ s```
 1. Why did you kill off `.. _core-howto-basics-twistd:`? Is there some rule to that?
 1. 25: `Applications` should be either `````Application``\ s``` or `applications`.
 1. 26: This line seems highly redundant to me.
 1. 27: Make this two sentences instead of an endash please.
 1. 28: the “(1)” is redundant. You could add it to line 17 and then never again.
 1. 30: JFK & Stalin
 1. 31: same as 5
 1. 37: What Adi said.
 1. 37: Could you add something like “If you’d like twistd to not start in the background and daemonize itself, you can pass it the `--nodaemon` option.”  I find this a very common use case nowadays (runit, upstart, supervisord…) that we shouldn’t make people search for it.
 1. 42: I hereby propose to kill off everything that comes after this line; `tap2rpm` and `tap2deb` should never be used ever again by anyone.  They only produce confused people on IRC.  I guess that’s a separate issue though.
 1. 45: `````Application`````

Feel free to merge after addressing these.  15 is obviously wishful thinking, 14 would make me happy but not vital.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl removed owner|
|-|-|

All done, except:


6 - It looked like a spurious comment that isn't use anywhere else.

I don't think that removing the rpm/deb thing would be that good. Maybe put in another ticket for that?

I added that extra line, but am putting up for review to make sure.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

for 6. it is not a spurious comment, but an anchor ID . I searched the code for 'core-howto-basics-twistd' and could not find any other place referring to it.
I think that it is safe to remove it. If sometime in the future, someone needs to refer to this exact paragraph from another file, the anchor ID can be re-created.

All good. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_7"></a>|@hynek set owner to @hawkowl|
|-|-|

1.25: there’s a spurious >
 1. 33: please add () to the functions (and only the functions)
 1. 34–37: hm, this repetition of `twistd` is gross.  Please make it an enumeration.
 1. 39: it should be possible to link to that directory, no?  I think Rich did that in his DNS client docs.
 1. 56: please remove the “easily” and open a ticket on tap2deb/tap2rpm deprecation.  Please search the tracker first, if something like that doesn’t exist already.

Please fix and re-submit for review, thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl commented|
|-|-|

Fixed, putting it up for review to make sure I did :)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_10"></a>|@adiroiban commented|
|-|-|

This looks good.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_11"></a>|@hynek commented|
|-|-|

Heya, sorry for the delay!

 - re:  tap2deb/tap2rpm-deprecation, what is the ticket number please?
 - re enumeration: that’s not exactly what I meant but it works too. :) A few nits:
  - Are the double commas around however really necessary?
  - If you want to capitalize the bullet points, don’t put commas at the end (but periods). :)
  - Are those scare-quotes around "forking to the background" really necessary?
  - “Automatically logging to a logfile,” sounds odd. How about something like “logging configuration” which is an internal link to http://twisted.readthedocs.org/en/latest/projects/core/howto/logging.html ?
  - How about making “And more.” a separate bullet and make it an internal link to http://twisted.readthedocs.org/en/latest/projects/core/howto/basics.html ?

Please address all of above and submit for last review. :)

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)|@hynek set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_13"></a>|@hawkowl removed owner|
|-|-|

Hi,

Made the changes.

However, with the last one, why would we want to make it an internal link to... itself? This is howto/basics.html :)

-hawkowl

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_14"></a>|@hynek commented|
|-|-|

I’m sorry, I meant http://twisted.readthedocs.org/en/latest/projects/core/howto/application.html – does that make sense to you?

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_15"></a>|@hynek commented|
|-|-|

Oh sorry didn’t see you put it into review again.

So I’d like to see the link above optionally added but otherwise LGTM.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_16"></a>|@hawkowl commented|
|-|-|

(In [43079]) Branching to basic-docs-format-6947-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_17"></a>|@hawkowl set status to `closed`|
|-|-|

(In [43082]) Merge basic-docs-format-6947-2: basics.rst should be properly formatted

Author: hawkowl
Reviewers: hynek, adiroiban
Fixes: [#6947](https://github.com/twisted/twisted/issues/6947)
